### PR TITLE
feat(tooltip): initial design guidance

### PR DESCRIPTION
**Pull request by naga-zone**
Added initial guidance for Tooltip.
[CDE-2343](https://vmw-jira.broadcom.net/browse/CDE-2343) - Create MD file for Tooltip Component.

### CODE REVIEW

File: 1035-tooltip-design-guidance.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+category: component design guidance
+state: draft
+created:2024-10-18
+updated:2024-10-21
```
Comment (kevinbuhmann) [inline 1035-tooltip-design-guidance.md:5]:
```suggestion
created: 2024-10-18
updated: 2024-10-21
```

File: 1035-tooltip-design-guidance.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+updated: 2024-10-22
+---
+
+# Tooltip Component
+
+[Tooltip](https://clarity.design/documentation/tooltip) component is a brief, informative message or short description about an element.
+
+## Guidance
+
+- **Must not** use tooltips on components. Instead, provide a descriptive label and use inline or signpost help for more information.
```
Comment (kevinbuhmann) [inline 1035-tooltip-design-guidance.md:14]: I'm not sure what this means.

File: 1035-tooltip-design-guidance.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+- **Must not** use tooltips on static images. Instead, use the HTML alt tag to provide information about the image.
+- **Must not** use tooltips on textual links. Instead, ensure that the link describes its destination and is not truncated.
+- **Should** use tooltips for actionable icons that do not have text labels, such as the icons in a toolbar. A tooltip is visible when the user hovers over the icon.
+- **Should** choose a [tooltip position]((https://clarity.design/documentation/tooltip#placement)) that ensures the entire tooltip is visible on the screen and the tooltip does not cover an important UI element.
```
Comment (kevinbuhmann) [inline 1035-tooltip-design-guidance.md:18]:
```suggestion
- **Should** choose a [tooltip position](https://clarity.design/documentation/tooltip#placement) that ensures the entire tooltip is visible on the screen and the tooltip does not cover an important page content.
```

File: 1035-tooltip-design-guidance.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- **Should** set a width that accommodates the text string.
+
+## Content Guidelines
+
+- **Must** use sentence-style capitalization and no ending punctuation.
+- **Must** use only plain text and be concise. A tooltip can be a sentence fragment. If more detailed information is required, use another form of help.
+- **Should** use a verb phrase to describe the action represented by the icon; for example, “Edit settings.”
```
Comment (kevinbuhmann) [inline 1035-tooltip-design-guidance.md:25]:
```suggestion
- **Should** use a verb phrase to describe the action represented by the icon; for example, “Edit settings”
```

Previous guidance says no ending punctuation.

File: 1035-tooltip-design-guidance.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+---
+category: component design guidance
+state: draft
+created: 2024-10-18
+updated: 2024-10-22
+---
+
+# Tooltip Component
```
Comment (kevinbuhmann) [inline 1035-tooltip-design-guidance.md:8]:
```suggestion
# Tooltip Design Guidance
```

File: 1035-tooltip-design-guidance.md (hunk 2 above)
Comment (naga-zone) [reply 1035-tooltip-design-guidance.md:14]: In my sense, you should add tooltip over any of the components like select/input component. Instead, have a label next to it and tag tooltip.

Comment (naga-zone) [reply 1035-tooltip-design-guidance.md:14]: This message is available at clarity.design as well.

File: 1035-tooltip-design-guidance.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+---
+category: component design guidance
+state: draft
+created: 2024-10-18
+updated: 2024-10-22
+---
+
+# Tooltip Design Guidance
+
+The [tooltip](https://clarity.design/documentation/tooltip) component facilitates display of a brief, informative contextual message or short description about an element.
```
Comment (jbusta5) [inline 1035-tooltip-design-guidance.md:10]:
```suggestion
The [tooltip](https://clarity.design/documentation/tooltip) component provides a brief, informative message or description that appears when a user hovers over or focuses on an element. Tooltips help clarify the purpose of buttons, icons, and other interactive elements without cluttering the interface.
```

File: 1035-tooltip-design-guidance.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+## Guidance
+
+- **Must not** use tooltips on components. Instead, provide a descriptive label and use inline or signpost help for more information.
```
Comment (jbusta5) [inline 1035-tooltip-design-guidance.md:14]:
```suggestion
- Design teams **must not** use tooltips on components. Instead, provide clear descriptive labels and use inline help or signposts for additional information. This approach ensures users receive essential context without the risk of missing important details that tooltips might present.
```

File: 1035-tooltip-design-guidance.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+- **Must not** use tooltips on static images. Instead, designs **should** call out alt text that provides a description of the image or state that the image is decorative (doesn't need alt text).
```
Comment (jbusta5) [inline 1035-tooltip-design-guidance.md:15]:
```suggestion
- Design teams **must not** use tooltips on static images. Instead, designs should provide alt text that describes the image or indicates that the image is decorative and does not require alt text. This ensures that all users, including those using screen readers, receive necessary information about the images, enhancing accessibility.
```

File: 1035-tooltip-design-guidance.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+- **Must not** use tooltips on textual links. Instead, ensure that the link describes its destination and is not truncated.
```
Comment (jbusta5) [inline 1035-tooltip-design-guidance.md:16]:
```suggestion
- Design teams **must not** use tooltips on textual links. Instead, links should clearly describe their destination without being truncated. This approach improves usability by providing users with immediate context about where the link will take them.
```

File: 1035-tooltip-design-guidance.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+- **Should** use tooltips for actionable icons that do not have text labels, such as the icons in a toolbar. A tooltip is visible when the user hovers over the icon.
```
Comment (jbusta5) [inline 1035-tooltip-design-guidance.md:17]:
```suggestion
- Design teams **should** use tooltips for actionable icons that lack text labels, such as those found in a toolbar. Tooltips provide essential context by displaying a brief description when the user hovers over the icon, helping users understand the icon's function without cluttering the interface with text labels.
```

File: 1035-tooltip-design-guidance.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+- **Should** choose a [tooltip position](https://clarity.design/documentation/tooltip#placement) that ensures the entire tooltip is visible on the screen and the tooltip does not cover an important page content.
```
Comment (jbusta5) [inline 1035-tooltip-design-guidance.md:18]:
```suggestion
- Design teams **should** choose a [tooltip position](https://clarity.design/documentation/tooltip#placement) that keeps the entire tooltip visible on the screen and ensures it doesn’t cover important page content. This positioning enhances usability by allowing users to access both the tooltip information and the relevant content without confusion or frustration.
```

File: 1035-tooltip-design-guidance.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+- **Should** set a width that accommodates the text string.
```
Comment (jbusta5) [inline 1035-tooltip-design-guidance.md:19]:
```suggestion
- Design teams **should** set a width for tooltips that accommodates the text string. This ensures that the tooltip is easily readable and prevents text from being cut off.
```

File: 1035-tooltip-design-guidance.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+
+## Content Guidelines
+
+- **Must** use sentence-style capitalization and no ending punctuation.
+- **Must** use only plain text and be concise. A tooltip can be a sentence fragment. If more detailed information is required, use another form of help.
```
Comment (jbusta5) [inline 1035-tooltip-design-guidance.md:24]:
```suggestion
- Design teams **must** use only plain text in tooltips, keeping the content concise. A tooltip can be a sentence fragment. If more detailed information is needed, consider using another form of help, such as a signpost or inline help, to ensure clarity and provide users with the necessary context.
```

File: 1035-tooltip-design-guidance.md (hunk 2 above)
Comment (naga-zone) [reply 1035-tooltip-design-guidance.md]: Jay has added more details. Hence resolving.

**Review by kevinbuhmann — APPROVED** [review]

**Pull request merged**